compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -135,6 +135,16 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             Transmute => {
+                assert!(src.layout.is_sized());
+                assert!(dest.layout.is_sized());
+                if src.layout.size != dest.layout.size {
+                    throw_ub_format!(
+                        "transmuting from {}-byte object to {}-byte object",
```

Comment:
```suggestion
                        "transmuting from {}-byte type to {}-byte type",
```
It might also be nice to print the types here